implementations/rust/ockam/ockam_command/src/service/start.rs

```diff
@@ -137,11 +138,12 @@ async fn start_service_impl<T>(
     serv_addr: &str,
     serv_name: &str,
     req: RequestBuilder<'_, T>,
+    tcp: Option<&'_ TcpTransport>,
 ) -> Result<()>
 where
     T: Encode<()>,
 {
-    let mut rpc = Rpc::background(ctx, opts, node_name)?;
+    let mut rpc = Rpc::background(ctx, opts, node_name, tcp)?;
```

Comment:
Instead of modifying the `Rpc::background` function, you have to use the `RpcBuilder`:

```rust
let mut rpc = RpcBuilder::new(ctx, opts, &cmd.node_opts.api_node).tcp(tcp)?.build();
```

Reply:
> Instead of modifying the `Rpc::background` function, you have to use the `RpcBuilder`:
> 
> ```rust
> let mut rpc = RpcBuilder::new(ctx, opts, &cmd.node_opts.api_node).tcp(tcp)?.build();
> ```

oh right!,  I didn't notice that  `tcp/1`  change the builder from `RpcMode::Embedded`  to `RpcMode::Background`  ,  I misread and assumed things created with the builder where always on `RpcMode::Embedded` mode.

Will modify :+1: